src/openklant/tests/test_migrate.py

```diff
@@ -689,3 +693,74 @@ def test_rsin(self):
         )
 
         self.assertEqual(output, [f"{self.live_server_url}{partij_url}"])
+
+    def test_klant_with_telefoonnummer_creates_digitaaladres(self):
+        # Klant with telefoonnummer creates a new DigitaalAdres
+        klant_with_phone = Klant(telefoonnummer="0612345678")
+
+        subject = Partij(nummer="12345", soort_partij=SoortPartij.persoon)
+        klant_with_phone._get_subject = lambda: subject
+
+        with patch(
+            "openklant.management.commands.migrate_to_v2_phonenumbers.OpenKlantClient"
+        ) as MockClient:
+            mock_client = MockClient.return_value
+            mock_client.retrieve.return_value = [
+                {"uuid": str(subject.uuid), "digitale_adressen": []}
+            ]
+            mock_client.create.return_value = {"uuid": "mock-digitaal-adres-uuid"}
+
+            StringIO()
+            results = _save_klanten_phonenumbers(
+                "http://localhost:8000", MagicMock(), [klant_with_phone]
+            )
+
+            self.assertIn("mock-digitaal-adres-uuid", results)
+            self.assertEqual(mock_client.create.call_count, 1)
+            digitaal_adres_data = mock_client.create.call_args[0][1]
+            self.assertEqual(digitaal_adres_data["adres"], "0612345678")
+            self.assertEqual(
+                digitaal_adres_data["soortDigitaalAdres"], "telefoonnummer"
+            )
+            self.assertEqual(digitaal_adres_data["referentie"], "portaalvoorkeur")
+
+        # Klant without telefoonnummer is skipped
+        klant_no_phone = Klant(telefoonnummer=None)
+        klant_no_phone._get_subject = lambda: subject
+
+        with patch(
+            "openklant.management.commands.migrate_to_v2_phonenumbers.OpenKlantClient"
+        ) as MockClient:
+            mock_client = MockClient.return_value
+            mock_client.retrieve.return_value = [
+                {"uuid": str(subject.uuid), "digitale_adressen": []}
+            ]
+            mock_client.create.return_value = {"uuid": "should-not-create"}
+
+            results = _save_klanten_phonenumbers(
+                "http://localhost:8000", MagicMock(), [klant_no_phone]
+            )
+            self.assertEqual(results, [])
+
+        # Klant with telefoonnummer but existing DigitaalAdres is skipped
+        klant_existing = Klant(telefoonnummer="0612345678")
+        klant_existing._get_subject = lambda: subject
+
+        existing_address = {
+            "soortDigitaalAdres": "telefoonnummer",
+            "referentie": "portaalvoorkeur",
+        }
+
+        with patch(
+            "openklant.management.commands.migrate_to_v2_phonenumbers.OpenKlantClient"
+        ) as MockClient:
+            mock_client = MockClient.return_value
+            mock_client.retrieve.return_value = [
+                {"uuid": str(subject.uuid), "digitale_adressen": [existing_address]}
+            ]
```

Comment:
Instead of mocking the v2 responses, you could use a `DigitaalAdresFactory` to create the telefoonnummer with portaalvoorkeur as referentie, and just let the call to the API be done like it would be done in other API unittests